src/app/Library/CrudPanel/Traits/Search.php

```diff
@@ -66,7 +66,7 @@ public function applySearchLogicForColumn($query, $column, $searchTerm)
                 case 'email':
                 case 'text':
                 case 'textarea':
-                    $query->orWhere($column['name'], 'like', '%'.$searchTerm.'%');
+                    $query->orWhere($query->getModel()->getTableWithPrefix().'.'.$column['name'], 'like', '%'.$searchTerm.'%');
```

Comment:
Since `$query->getModel()->getTableWithPrefix().'.'.smth` is done in all cases of the switch statement, I propose we get the table name above the switch statement and use `$tableName` inside the Switch.

```php
$tableName = $query->getModel()->getTableWithPrefix();
```

Reply:
@tabacitu this would not work for `select_multiple` where we need to get the `connected model` and only there get the table name. 

You still think we should do this even if it does not work the same way for all columns ?

Reply:
Hmmm right. In that case maybe we should add a new method on `CrudTrait` that will accept the column as a parameter, and return the column prefixed with the table name (and table prefix if any). So something like:

```php
$query->getModel()->getColumnNameWithTableName($column)
// or
$query->getModel()->getFullColumnName($column)
// or
$query->getModel()->getNamespacedColumnName($column)
```

I'm wondering though... isn't there already an Eloquent method for this? Sounds like something that should already be handled by Eloquent...

Reply:
Hey @tabacitu I don't think eloquent have a function for that. The prefix is set in the database connection config, not in model, so `getTable()` returns without prefix. But if you setup a `prefix_` in your connection config when calling `Model::all()` eloquent will translate to: `SELECT * FROM prefix_table`. 

As we are building the query if we only `->where('attr', 'smt')` the query is translated to `SELECT ... WHERE attr = 'smt'`, if we join a table with the same `attr` name we know where this is going to end. 🙃 

I agree we could setup this in a function like you propose. I would go with `$query->getModel()->getColumnWithTableName()`

What you think ? 

Best,
Pedro